mlrun/serving/routers.py

```diff
@@ -157,3 +163,256 @@ def _handle_event(self, event):
         response = route.run(event)
         event.body = response.body if response else None
         return event
+
+
+class VotingEnsemble(ModelRouter):
+    """Voting Ensemble class
+
+        The `VotingEnsemble` class enables you to apply prediction logic on top of
+        the different added models.
+
+        You can use it by calling:
+        - <prefix>/<model>[/versions/<ver>]/operation
+            Sends the event to the specific <model>[/versions/<ver>]
+        - <prefix>/operation
+            Sends the event to all models and applies `vote(self, event)`
+
+        The `VotingEnsemble` applies the following logic:
+        Incoming Event -> Router Preprocessing -> Send to models (or model) ->
+        Apply all model logic (Preprocessing -> Prediction -> Postprocessing) ->
+        Router Voting logic -> Router Postprocessing -> Response
+
+        This enables you to do the general preprocessing and postprocessing steps
+        once on the router level, with only model-specific adjustments at the
+        model level.
+    """
+
+    def __init__(
+        self,
+        context,
+        name,
+        routes=None,
+        protocol=None,
+        url_prefix=None,
+        health_prefix=None,
+        vote_type=None,
+        executor_type=None,
+        **kwargs,
+    ):
+        super().__init__(
+            context, name, routes, protocol, url_prefix, health_prefix, **kwargs
+        )
+        self.url_prefix = url_prefix or f"/{self.protocol}/models"
+        self.name = "VoteRouter"
```

Comment:
Are you sure the name should be constant ? if so, then why you're getting it in the args ?
maybe `self.name = name or "VoteRouter"`

Reply:
taking name from class with default